### ISSUE TITLE
General Suggestions For Docs

### ISSUE THREAD

**Issue by joshkestenberg** [tendermint/tendermint.github.io#51]
1. Flow chart at https://tendermint.com/intro/abci-overview is really vague. No very starting point and it could also be interpreted as a clockwise circular flow all the way around as opposed to a series of request/response cycles. Mempool should be the highest point in the diagram (in this kind of chart, top left doesn't necessarily feel like the beginning) and request/response pairs should be colour-coded or numbered.  Diagrams from 23:15 to 25:50 with accompanied explanation in this video (http://www.youtube.com/watch?v=9krAQzN6tbc&t=23m16s) clarified things for me far more than this flow chart.

2. Flow chart at https://tendermint.com/intro/consensus-overview is a lot more obvious, but I'd still suggest putting 'new height' as the highest point in the diagram, as opposed to top left.

3. On Docs home (https://tendermint.com/docs), this line: "To learn more about Tendermint’s various pieces, checkout the Documentation." 
- 'checkout' should be 'check out' --- 2 words.
-'documentation' links to the same page I'm already on (https://tendermint.com/docs)--- redundant.

4. https://tendermint.com/docs/guides/app-development has same vague diagram as (1). Also, the 'message types' material here (https://github.com/tendermint/abci) should be displayed here, or at least linked to here. I also think the bullet point section on this page should be reformatted (bold primary points, better spacing, etc). It's kind of hard to look at. I also think that this page should be broken up into components. It reads more like a glossary than a guide, and the formatting is all uniform (eg. gRPC and TSP should be sub-categories of Server, Connections should be a section -or a page-, Messages should be a section, etc.)

5. All guides could use some navigation, be it in the side bar, or at the top of the actual page. Some of them are long and it'd be useful to be able to see a list of topics and just pick the one that I'm looking for.

**Comment by zramsay** [comment]
Some additional suggestions:
1. The intro should focus exclusively on, well, introducing tendermint. It should be readable/understandable by any non-programmer and any reference to code snippets removed. "Code" in this sense belongs (IMO) exclusively in the  "Docs" sections. The "Getting Started" sidebar section from the "Intro" should be moved into the "Docs" section. Along those lines, all explainer material from the "Docs" section should be re-worked into the "Intro". To sum: "Intro" should have no code & be focused towards explaining tendermint to non-programmers whereas the "Docs" assume a basic knowledge of tendermint (i.e., having read the intro) and is for programmers only (so docs should be code only).

2. The "Internal" sidebar section from the "Docs" should either be collapsed into a single-file or somewhat hidden. AFAICT, the content from this section is for a fairly advanced users & as a beginner reading the docs, it was confusing/additional cognitive overhead/distracting without adding any value with respect to "how to do  X".

3. The dummy app appears out of nowhere in https://tendermint.com/docs/guides/abci-cli (i.e., where the heck did that binary come from?!) and its not clear what the dummy app is meant to do/what it integrates. After implementing the other suggestions mentioned here, that would be the first thing to tackle.

4. I'm not a fan of "Guides" sections at all. The first part is a tutorial, the second and third and have to do with theory/architecture of tendermint while the last three have to do with contributing. The "Docs" section should focus on 1) Getting Started (moved from the Intro section) and 2) a couple good tutorials. "Guides" needs to go.

5. All the above suggestions should harmonize in a coherent place for the documentation. The `docs/` directory in tendermint/tendermint should hold all relevant documentation files and be cleaned up (ADR stuff removed & possibly put in a repo/directory named RFCS because thats what they are. The use of a `docs/` directory is so strong a standard that github has implemented automated documentation from that directory itself.

6. To address some the changes mentioned, the "Docs" section should distinguish between practical tutorials for developers and theory/architecture/internals/advanced (so basically two sections, where the latter can be avoided by anyone getting started)

**Comment by melekes** [comment]
Agreed with almost all points except

> should have no code & be focused towards explaining tendermint to non-programmers

It depends on who is our target auditory. if it is programmers, then it is ok. for example consul is a tool for devs and [getting started is a part of the intro](https://www.consul.io/intro/index.html).

> The "Internal" sidebar section from the "Docs" should either be collapsed

collapsed by default? again https://www.consul.io/docs/index.html for example

> The dummy app appears out of nowhere in https://tendermint.com/docs/guides/abci-cli (i.e., where the heck did that binary come from?!)

https://tendermint.com/intro/getting-started/first-app

**would be nice to have the ability to link to paragraphs (i.e. `something#part1`)**

> "Guides" needs to go.

go where? agree that some things could be changed, but removing it?

> I'm not a fan of "Guides" sections at all. 

providing examples of docs which do things differently to **back up your opinion** would be great (something like `X has great docs and they do Y`).

**I think guides/contributing should be moved to the repo**
**guides/install-go should be removed. why duplicate when https://golang.org/doc/install does a perfect job?**

**Comment by zramsay** [comment]
right, good points. So I've assumed that many non-programmers land on the tendermint site just wanting to learn more about tendermint and not necessarily use the tools. That assumption could of course be wrong. That said, the biggest challenge I found navigating the site as a new user of tendermint was the lack of flow. The intro has content that is documentation and the documentation has content that is introductory.

WRT to the "Internals" sidebar - this again could be a flawed assumption on my part. I've assumed that the large majority of devs coming to the docs site don't care about the internals, hence why I'd "hide" it away in an advanced section or something (like within `docs/specs/`)

Ah, so that's where the dummy app is explained. This is quite confusing, because I went straight to the `/docs` page (as developers tend to do! [maybe another wrong assumption]) and now to get relevant information, I'd have to go back to the `/intro` (but there's no link back to the getting-started page; something like: "This guide/tutorial assumes you've read the [getting-started page](link)"

Markdown supports links-to-headers, which basically accomplishes what you suggested for paragraphs.

Guides - I essentially meant the concept should be deprecated (replaced with "Tutorials"). Like you've said, we can deduplicate the go installation, and everything else can be shuffled around to either the repo or a tutorial.
